Assets/VRTK/Scripts/VRTK_PlayerJump.cs

```diff
@@ -0,0 +1,366 @@
+        private float crouchLowTime;
+
+        // velocity history
+        private float[] historicVelocity;
+
+        // number of history points to compare
+        private int historyLen = 5;
+
+        private Transform HMDTransform;
+        private VRTK_PlayerPresence playerPresence;
+
+        #endregion
+
+        #region ### private logging vars ###
+
+        // set to writable directory eg. C:/Users/Public/Documents/Unity Projects/MyProject/Assets
+        private string loggingTarget = "";
+        private StreamWriter fileWriter;
+        private bool logging = false;
+        private bool debugging = true;
+
+        #endregion
+
+        private void Awake()
+        {
+            // Required Component: VRTK_PlayerPresence
+            playerPresence = GetComponent<VRTK_PlayerPresence>();
+            if (!playerPresence)
+            {
+                playerPresence = gameObject.AddComponent<VRTK_PlayerPresence>();
+            }
+            playerPresence.SetFallingPhysicsOnlyParams(true);
+            playerPresence.StartPhysicsFall(Vector3.zero);
+
+            HMDTransform = VRTK_DeviceFinder.HeadsetTransform();
+
+            historicVelocity = new float[historyLen];
+        }
+
+        private void Start()
+        {
+            baselineHeight = HMDTransform.position.y;
+            // TODO - may want a cooldown timer to prevent spurious jumps when headset is first put on
+            crouchLowTime = -1;
+
+            if (logging)
+            {
+                InitLogging();
+            }
+        }
+
+        private void Update()
+        {
+            CheckPosture();
+        }
+
+        private void CheckPosture()
+        {
+            float jumpVelocity = 0;
+
+            var dv = UpdateHeightHistoryAndReturnVelocity(HMDTransform.position.y);
+
+            if (jumping)
+            {
+                CheckForJumpEnd(dv);
+            }
+            else
+            {
+                UpdateBaseline(dv);
+                CheckForCrouchStart(dv);
+
+                if (CrouchWasDetected())
+                {
+                    UpdateCrouchLowpoint();
+
+                    if (CrouchTimerNotExpired())
+                    {
+                        if (VelocityChangeAboveThreshold(dv))
+                        {
+                            // if head position exceeds the baseline
+                            if (HMDHeightExceedsBaseline())
+                            {
+                                jumping = true;
+
+                                jumpVelocity = jumpFixed ? 1 : CalcAdjustedFinalJumpVelocity();
+
+                                DoJump(jumpVelocity);
+                            }
+                        }
+                    }
+                    else
+                    {
+                        crouchStartTime = 0;
+                        crouchLowTime = Time.time;
+                        crouchLow = baselineHeight + 1000000; // TODO probably there is a better way
```

Comment:
worth replacing this number with a variable with a meaningful name so you can understand what the 100000 is for.